libnmstate/nm/profile.py

```diff
@@ -417,11 +418,12 @@ def _generate_connection_settings(self, ifname):
         original_state_wired = {}
         if self._iface.is_desired:
             original_state_wired = self.original_iface_info
-        wired_setting = wired.create_setting(
-            original_state_wired, self._remote_conn
-        )
-        if wired_setting:
-            settings.append(wired_setting)
+        if self.iface.type != InterfaceType.INFINIBAND:
+            wired_setting = wired.create_setting(
+                original_state_wired, self._remote_conn
+            )
+            if wired_setting:
+                settings.append(wired_setting)
```

Comment:
I do not understand why is this needed. Could you explain it?

Reply:
Included these comments:
  
          # The IP over InfiniBand has its own setting for MTU and does not
            # have ethernet layer.